src/scripts/make-webpack-config.ts

```diff
@@ -22,11 +22,10 @@ interface AliasedConfiguration extends Configuration {
 	resolve: Resolve & { alias: Record<string, string> };
 }
 
-export default function (
-	config: Rsg.SanitizedStyleguidistConfig,
-	env: 'development' | 'production' | 'none'
-): Configuration {
-	process.env.NODE_ENV = process.env.NODE_ENV || env;
+type Mode = Configuration['mode'];
+
+export default function (config: Rsg.SanitizedStyleguidistConfig, env: Mode): Configuration {
+	process.env.NODE_ENV = (process.env.NODE_ENV as Mode) || env;
```

Comment:
I think it should be something like this:

```suggestion
export default function (config: Rsg.SanitizedStyleguidistConfig, envRaw: Mode): Configuration {
	const env = (process.env.NODE_ENV as Mode) || envRaw;
```

Reply:
We can try, I will do the commit from here